COMETwebapp/Primitives/BasicPrimitive.cs

```diff
@@ -133,48 +142,81 @@ public void ResetRotation()
+        public Task SetPositionFromElementUsageParameters(ElementUsage elementUsage, Option selectedOption, List<ActualFiniteState> states)
         {
-            const string shortName = "coord";
+            IValueSet? valueSet = null;
+
+            valueSet = this.GetElementUsageValueSet(elementUsage, selectedOption, states, Scene.PositionShortName);
+
+            if (valueSet is not null &&
+                double.TryParse(valueSet.ActualValue[0], out var x) &&
+                double.TryParse(valueSet.ActualValue[1], out var y) &&
+                double.TryParse(valueSet.ActualValue[2], out var z))
+            {
+                this.SetTranslation(x / 1000.0, y / 1000.0, z / 1000.0);
+            }
 
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Gets the value sets asociated to an element usage depending on the selected option and the available states
+        /// </summary>
+        /// <param name="elementUsage">the <see cref="ElementUsage"/> used for query the value set</param>
+        /// <param name="selectedOption">the current <see cref="Option"/> selected</param>
+        /// <param name="states">The available states</param>
+        /// <returns></returns>
+        protected IValueSet? GetElementUsageValueSet(ElementUsage elementUsage, Option selectedOption, List<ActualFiniteState> states, string parameterTypeShortName)
+        {
             ParameterBase? parameterBase = null;
             IValueSet? valueSet = null;
 
             if (elementUsage.ParameterOverride.Count > 0)
             {
-                parameterBase = elementUsage.ParameterOverride.FirstOrDefault(x => x.ParameterType.ShortName == shortName
-                                                        && x.ParameterType is CompoundParameterType);
-            }
-            else
+                parameterBase = elementUsage.ParameterOverride.FirstOrDefault(x => x.ParameterType.ShortName == parameterTypeShortName
+                                                                                   && x.ParameterType.GetType() == Scene.GetParameterTypeFromParameterShortName(parameterTypeShortName));
+            }            
+            
+            if(parameterBase is null)
             {
-                parameterBase = elementUsage.ElementDefinition.Parameter.FirstOrDefault(x => x.ParameterType.ShortName == shortName
-                          && x.ParameterType is CompoundParameterType);
+                parameterBase = elementUsage.ElementDefinition.Parameter.FirstOrDefault(x => x.ParameterType.ShortName == parameterTypeShortName
+                                                                                             && x.ParameterType.GetType() == Scene.GetParameterTypeFromParameterShortName(parameterTypeShortName));
             }
 
             if (parameterBase is not null)
             {
-                foreach (var actualFiniteState in states)
+                if(states.Count > 0)

```

Comment:
the `Count` is not required, of the list is empty the for each will not hit, please remove